data-science-platform/examples/gpu-sd-iaas-jupyter/index.md

```diff
@@ -0,0 +1,152 @@
+5. Create a GPU enabled virtual machine by clicking Instances > Launch instance, and
+    1. In Details, Instance name: Put a good name for a compute server, like "Jupyter demo".
+    2. In Source, click the up arrow icon next to an image that has Docker, CUDA, Miniforge, Jupyter Lab and RMD.
+    3. In Flavor, click the up arrow icon next to a flavor that has GPU. Bigger is more expensive.
+    4. If applicable: In Security Groups, click the up arrow icon next to allowall.
+    5. In Key Pair, verify that your key is allocated.
+    6. Click Launch instance.
+6. Click Associate Floating IP > IP Address > pick one, fill in in next step.
+7. Wait for Power State to become Running.
+
+### 2. Configure your computer for easy access.
+
+Add to SSH-config (eg `~/.ssh/config`):
+
+```ssh
+Host jupyter-demo
+  HostName [associated IP in Horizon]
+  User ubuntu
+  ProxyJump sshuser@dsp.aida.scilifelab.se
+  ServerAliveInterval 10
+  LocalForward 8888 localhost:8888
+  LocalForward 6006 localhost:6006
+```
+
+This sets up your computer use the DSP SSH access gateway when making SSH
+connections to your VM. By default, DSP rejects SSH connections that are not
+made through the SSH access gateway.
+
+**Note**: In the future, we will switch to `ProxyJump your.email@example.com@dsp.aida.scilifelab.se`
+to ensure that users will only be able to connect to their own secure
+environments.
+
+ServerAliveInterval makes it easier to maintain a connection, and to detect when
+it has gone stale.  
+
+The LocalForwards define SSH secured port forwards. They connect ports on your
+computer with ports on your VM in the secure environment. They allow you to work
+with Jupyter notebooks and TensorBoard that are running on the VM in your secure
+environment as if though they were running on your computer.
+
+### 3. Install software from public repositories that are trusted by the platform.
+
+DSP secure environments block connections by default. However, DSP provides an
+inspecting http proxy that enables downloading software and security updates
+from public repositories that are trusted by AIDA Data Hub. DSP data science
+images are preconfigured to make transparent use of this proxy, as demonstrated
+in this next step.
+
+Here, clone the Jupiter notebook GitHub repo and use apt and pip to install its
+dependencies in a Python virtual environment:
+
+```bash
+ssh jupyter-demo
+git clone https://github.com/eryl/aida-transformers-workshop-code.git
+cd aida-transformers-workshop-code
+sudo apt install python3-venv
+python3 -m venv .venv
+source .venv/bin/activate
+pip install -r requirements.txt
+```
+
+**Note**:
+The restrictivity of the DSP inspecting http proxy is continually updated, to
+adjust to updates in public repositiories that make them more or less
+appropriate for secure environments. For example, publicly accessible granular
+download counters are increasingly popular despite constituting a trivially
+exploitable data exfiltration method.
```

Comment:
Would it be of public interest to point to an article which explains this data exfiltration? At least for me it was an eye-opener.

Reply:
I don't think it's needed here. For later, I hope we can offer PIs the choice of what to allow with help text and if dare, our evaluation of associated risks. Those should definitely have read more bits (and the same information without the ability to choose should likely be available to users).

Reply:
Somewhat related; if there are large scale data sources (something people often want to use and would be generally known), we probably want to consider them for access as well.

Reply:
Re: data sources, we can consider adding those on a case by case basis on request.